### PR TITLE
chore: add regex in `InlineEditor` [DET-7518]

### PR DESCRIPTION
**Pull request by keita-determined**
## Description
[DET-7518]

## Screenshot

https://user-images.githubusercontent.com/109113616/182768603-4b33765a-742e-45dd-a44f-5c92fffb384c.mov

## Test Plan
- Check if InlineEditor works correctly
- Check if display name editor works correctly 
  - should shake when its empty or non alphanumeric is used, and previous value should be set
  - should save data when input is alphanumeric

## Commentary (optional)
Jira ticket says 
> When a user edits their displayName to be empty, it should get set to null

but decided that empty value is invalid

## Checklist

- [ ] User-facing API changes need the "User-facing API Change" label.
- [ ] Release notes should be added as a separate file under `docs/release-notes/`.
See [Release Note](https://github.com/determined-ai/determined/blob/master/docs/release-notes/README.md) for details.
- [ ] Licenses should be included for new code which was copied and/or modified from any external code.
- [ ] If modifying `/webui/react/src/shared/` verify `make -C webui/react test-shared` passes.

[DET-7518]: https://determinedai.atlassian.net/browse/DET-7518?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DET-1234]: https://determinedai.atlassian.net/browse/DET-1234?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: webui/react/src/components/InlineEditor.module.scss [hunk 1]
```diff
@@ -111,3 +111,26 @@
 .base.disabled .growWrap:hover > textarea {
   border-color: transparent;
 }
+.shakeAnimation {
```
Comment (hkang1) [inline webui/react/src/components/InlineEditor.module.scss:114]: ha! nice little animation, love it

File: webui/react/src/components/InlineEditor.module.scss [hunk 2]
```diff
@@ -111,3 +111,26 @@
...
+  animation: shake 0.8s cubic-bezier(0.36, 0.07, 0.2, 0.95) both;
```
Comment (hkang1) [inline webui/react/src/components/InlineEditor.module.scss:115]: **question:** would it be possible (and do you think it'll be a good idea) to change the border to `var(--theme-status-critical)` as it's shaking and back to the regular border color to show that it's a problem?

File: webui/react/src/pages/Settings/SettingsAccount.tsx [hunk 3]
```diff
@@ -57,6 +57,7 @@ const SettingsAccount: React.FC = () => {
       <div className={css.row}>
         <label>Display Name</label>
         <InlineEditor
+          pattern={new RegExp('^[a-zA-Z]+[a-zA-Z0-9]*$')}
```
Comment (hkang1) [inline webui/react/src/pages/Settings/SettingsAccount.tsx:60]: idk if there is a significant performance difference, but could also be written as:
```
new RegExp('^[a-z]+[a-z0-9]*$', 'i')
```
OR (this one is succinct but may not be as typescript friendly? not sure)
```
/^[a-z]+[a-z0-9]*$/i
```

**Review by hkang1 — APPROVED** [review]
This looks great! I added some minor suggestions and comments, but nothing blocking.

File: webui/react/src/pages/Settings/SettingsAccount.tsx (hunk 3 above)
Comment (keita-determined) [reply webui/react/src/pages/Settings/SettingsAccount.tsx:60]: That's a good suggestion. Enabling insensitivity improves the performance a lot in my experience

File: webui/react/src/components/InlineEditor.module.scss (hunk 2 above)
Comment (keita-determined) [reply webui/react/src/components/InlineEditor.module.scss:115]: I think its possible and a good idea. My concern is that if the entered value is invalid, the previous value is set immediately. Ideally setting the previous value after shaking would be better with the `critical border`. Let me try

Comment (keita-determined) [reply webui/react/src/components/InlineEditor.module.scss:92]: I fixed it. It was actually not hard to do

https://user-images.githubusercontent.com/109113616/183133976-92a3f6bc-c25d-485d-b5e1-c519040ba77c.mov

File: webui/react/src/pages/Settings/SettingsAccount.tsx [hunk 4]
```diff
@@ -58,6 +58,7 @@ const SettingsAccount: React.FC = () => {
         <label>Display Name</label>
         <InlineEditor
           maxLength={32}
+          pattern={new RegExp('^[a-z][a-z0-9\\s]*$', 'i')}
```
Comment (keita-determined) [inline webui/react/src/pages/Settings/SettingsAccount.tsx:61]: Forgot to add whitespace, so added

**Pull request merged**